### PR TITLE
osutil: add package for reading Build-ID

### PR DESCRIPTION
**Pull request by zyga**
*Lets make sure we merge this via a "squash" merge - we have some binary data from the previous iteration of the tests in here.*

This patch adds a small package that reads the GNU Build-ID note from an
elf executable. Build-ID is a standard feature in many distributions,
eixsting ever since Fedora 8 introduced it. It facilitiates crash
analysis as each crash contains identified unique for each build that
can be correlated to debugging symbols.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

File: osutil/build_id/build_id_test.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+func (s *buildIDSuite) TestGetBuildID(c *C) {
+	for _, t := range []struct {
+		fname, expected string
+	}{
+		{"true.amd64", "BuildID[sha1]=efbf0ce8dd9617c890a0547ce5a1a6073f5867af"},
+		{"true.arm64", "BuildID[sha1]=8b65339d7fa0c4cdc87ed9c8020626aa10fb521b"},
+		{"true.armhf", "BuildID[sha1]=c80229c22d4b6b30b71ab1b1b5a1de6b86b6aadf"},
+	} {
+		id, err := bi.GetBuildID(t.fname)
```
Comment (zyga) [inline osutil/build_id/build_id_test.go]: I guess I need to teach the test to find the test data ... ideas welcome.

**Comment by zyga** [comment]
Thank you for your review @mwhudson! I was aware of the suggested way to find notes and I guess it doesn't hurt to just scan all note sections. As for alignment, thanks a lot, I was not aware note fields are aligned like that. Can you perhaps tell me if that is expressed in the `sh_addralign` field of the section header? I noticed it was 0x04 but I didn't connect it to alignment inside the variable-sized fields.

**Comment by zyga** [comment]
@mwhudson I've updated the code to scan all sections as you suggested. Can you please render some advice on alignment. I _think_ we are OK (as you said, both elements we read are naturally aligned) but I wanted to understand how to do it correctly.

File: osutil/build_id.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+// ErrNoBuildID is returned when an executable does not contain a Build-ID
+var ErrNoBuildID = errors.New("executable does not contain GNU Build-Id")
+
+// BuildID is an array of bytes that identify a given build of an executable.
+type BuildID []byte
+
+// String returns the familiar representation, namely "BuildID[...]=..."
+func (id BuildID) String() string {
+	switch len(id) {
+	case 0x14: // SHA1 note:
```
Comment (chipaca) [inline osutil/build_id.go]: 0x10 would be MD5 or UUID (per the fedora doc), which in `file` is printed as `"md5/uuid"`; I think we should do the same

**Comment by mvo5** [comment]
Lets make sure we merge this via a "squash" merge - we have some binary data from the previous iteration of the tests in here.

**Comment by zyga** [comment]
This branch is not for 2.22.6 but we'll cherry-picked the squashed commit to 2.22.6 later.

**Pull request merged**